### ISSUE TITLE
Arguments do not match for generic procedure ```max```

### ISSUE THREAD

**Issue by Pranavchiku** [lfortran/lfortran#670]
on executing ```./src/bin/lfortran --show-asr --implicit-typing --fixed-form ../scipy/scipy/optimize/minpack2/dcstep.f```
throws the following error 
```
semantic error: Arguments do not match for any generic procedure, max
   --> ../scipy/scipy/optimize/minpack2/dcstep.f:112:14
    |
112 |          s = max(abs(theta),abs(dx),abs(dp))
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that needs to be fixed).
```

**Comment by Pranavchiku** [comment]
I am currently not able to find the path from where this error is originating, will try to do that asap

**Comment by Pranavchiku** [comment]
Resolved in #669

**Comment by Smit-create** [comment]
Fixed in #675
